pages/_app.tsx

```diff
@@ -1,6 +1,44 @@
-import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import "../styles/globals.css";
+import type { AppProps } from "next/app";
+import { useEffect, useState } from "react";
+import { Criteria, Member } from "../interfaces/interfaces";
 
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [desiredPlayer, setDesiredPlayer] = useState<Criteria>({
+    skill: "Anything will do",
+    language: "Anything will do",
+    gender: "Anything will do",
+  });
+
+  const [allPlayers, setAllPlayers] = useState<Member[]>([]);
+
+  async function getAllPlayers(): Promise<void> {
+    const response = await fetch("/api/allPlayers");
+    if (!response.ok) {
+      return console.error(
+        "Error with the response of the fetch. Response status: ",
+        response.status
+      );
+    } else {
+      const listOfAllPlayers = await response.json();
+      setAllPlayers(listOfAllPlayers);
+    }
+  }
```

Comment:
You should wrap this code in a `try / catch` block to handle unexpected errors (no internet, server down etc).

Reply:
Thank you, I forgot about it